### ISSUE TITLE
Check the expression for Hdotdot in notes and codes

### ISSUE THREAD

**Issue by EFTCAMB** [EFTCAMB/EFTCAMB#15]
Changing Hdotdot for full compliance with the notes does not respect legacy results. Fix it.

**Comment by simonepeirone** [comment]
I am a bit concerned about the notation.
grhom_t refers to total matter density, it should then be the same as \rho_m + \rho_nu (using notes notation). Where \rho_m  includes non rel. matter, dark matter, radiation and massless neutrinos and  \rho_nu is the density contribute of the massive neutrinos.

If this is the case, the correct expression for Hdotdot should be:

eft_cache%Hdotdot = eft_cache%adotoa_( (eft_cache%grhom_t -eft_cache%grhonu_tot)/6.0_dl +( eft_cache%gpresm_t -eft_cache%gpinu_tot) +1.5_dl_( eft_cache%gpresm_t -eft_cache%gpinu_tot )**2/( eft_cache%grhom_t -eft_cache%grhonu_tot ) + ... (the following terms look all right)

**Comment by EFTCAMB** [comment]
The part of code that is responsible is:

```
eft_cache%Hdotdot = 2._dl*eft_cache%adotoa*eft_cache%Hdot &
            & + 0.5_dl*eft_cache%adotoa*(eft_cache%grhob_t + eft_cache%grhoc_t + 8._dl*(eft_cache%grhog_t+eft_cache%grhor_t)/3._dl)&
            & + 0.5_dl*eft_cache%adotoa*eft_cache%grhov_t*( (1._dl+self%PureEFTwDE%value(a) )*(1._dl+3._dl*self%PureEFTwDE%value(a)) -a*self%PureEFTwDE%first_derivative(a))&
            & + eft_cache%adotoa/6._dl*eft_cache%grhonu_tot -0.5_dl*eft_cache%adotoa*eft_cache%gpinu_tot -0.5_dl*eft_cache%gpinudot_tot
```

where:

```
H=adotoa
grhoX_t = rho_X*a^2/m_0^2
```

**Comment by EFTCAMB** [comment]
The one that instead complies with numerical notes is:

```
eft_cache%Hdotdot = eft_cache%adotoa*( ( eft_cache%grhob_t +eft_cache%grhoc_t)/6._dl +2._dl*( eft_cache%grhor_t +eft_cache%grhog_t)/3._dl ) &
        & +eft_cache%adotoa*eft_cache%grhov_t*( 1._dl/6._dl +self%PureEFTwDE%value(a) +1.5_dl*self%PureEFTwDE%value(a)**2 -0.5_dl*a*self%PureEFTwDE%first_derivative(a) ) &
        & +eft_cache%adotoa*eft_cache%grhonu_tot/6._dl -0.5_dl*eft_cache%adotoa*eft_cache%gpinu_tot -0.5_dl*eft_cache%gpinudot_tot
```

Which one is correct?

**Comment by simonepeirone** [comment]
From the code I see:

`eft_cache%grhom_t  = eft_cache%grhob_t +eft_cache%grhoc_t +eft_cache%grhor_t +eft_cache%grhog_t +eft_cache%grhonu_tot`

meaning that we consider in grhom_t also the contribution from massive neutrinos(grhonu_tot). However, this is not true in the numerical notes, where it is clear that rho_m does not include the contribution from rho_nu, but only baryons, CDM, radiation and massless neutrinos.

Said so, I think the most convenient way of writing this equation is to consider directly the expression for grhom_t 

`eft_cache%Hdotdot = eft_cache%adotoa*( (eft_cache%grhom_t -eft_cache%grhonu_tot)/6.0_dl +( eft_cache%gpresm_t -eft_cache%gpinu_tot) +1.5_dl*( eft_cache%gpresm_t -eft_cache%gpinu_tot )**2/( eft_cache%grhom_t -eft_cache%grhonu_tot )) + eft_cache%adotoa*eft_cache%grhov_t*( 1._dl/6._dl +self%PureEFTwDE%value(a) +1.5_dl*self%PureEFTwDE%value(a) **2 -0.5_dl*a*self%PureEFTwDE%first_derivative(a) )  +eft_cache%adotoa*eft_cache%grhonu_tot/6._dl -0.5_dl*eft_cache%adotoa*eft_cache%gpinu_tot -0.5_dl*eft_cache%gpinudot_tot`

**Comment by simonepeirone** [comment]
I just checked the difference between these two expression for Hddot 
![hdotdot_deviation](https://cloud.githubusercontent.com/assets/17548561/20463547/cc623808-af35-11e6-96fe-31e001b0f987.png)

In this plot you see the fractional deviation between the old expression (Hdotdot_2) and the new one (Hdotdot_1). 
I reported here this deviation in two cases: when massive neutrinos are not present (omnuh2=0, the red line) and when we consider massive neutrinos (omnuh2=0.2, the green line).
As we were suggesting, the difference between these two expressions is present only when massive neutrinos are taken into account.
When omnuh2=0 (the red line), this difference is equal to zero, up to 15 digits.
